### ISSUE TITLE
Cloud Trace API Support

### ISSUE THREAD

**Issue by iamacarpet** [a1comms/GaeSupportLaravel#3]
Google are in the process of merging a new interface to interact with Cloud Trace GoogleCloudPlatform/google-cloud-php#456

Once their PR is merged into master, we need to introduce Trace support here.

We'll start out with the Laravel provider here: https://github.com/chingor13/php-test-apps/blob/master/laravel/blog/app/Providers/GoogleCloudProvider.php

Keep it using AsyncReporter and QpsSampler on Flexible, but modify it so it'll detect the Standard Environment and revert back to our old method of using Push Queues to do an async submit as the latency is a lot lower, plus use the AlwaysOffSampler, as we only want to sample when App Engine wants us to which it'll signal by sending `o=1` in the header.

To this end, we'll need to write our own Reporter interface that'll utilise the Push Queue API.

Since using the Push Queue method will use more chargeable resource, we should publish our cost tuned Go microservice that can be deployed to do the async work from this library.

Although, should we write the URI handlers to do the same work in here for people who don't want to utilise this method? Perhaps if they are super low volume and find it easier to stay within the free tier that way?

**Comment by iamacarpet** [comment]
Google have merged the Trace API support into GoogleCloudPlatform/google-cloud-php#456

I have merged some initial Trace support into this library, but there is an issue with the Standard Environment, which is open at GoogleCloudPlatform/google-cloud-php#526

To use the Trace support, include the service provider:

```php
A1comms\GaeSupportLaravel\Trace\GoogleCloudTraceProvider::class
```

Then use the `Google\Cloud\Trace\RequestTracer` class to create Trace Spans.
